### ISSUE TITLE
Lens becomes unusably laggy in a large cluster

### ISSUE THREAD

**Issue by Double0negative** [lensapp/lens#2243]
**Describe the bug**
Lens becomes unusably laggy with a large cluster, to the point where it becomes completely frozen if the size grows enough. I am attempting to use it on a cluster with ~20k pods (many hundreds starting/stopping per minute) and the app completely hangs and never recovers when attempting to do anything (especially the pods tab). 

**To Reproduce**
1. Have significantly large cluster and connect
2. open pods tab
3. Lens will freeze after a few seconds

**Expected behavior**
Lens doesn't freeze.

**Environment (please complete the following information):**
- Lens Version: 4.1.3
- OS: [e.g. OSX]

**Comment by jim-docker** [comment]
Are these pods all in a single namespace?

**Comment by Double0negative** [comment]
No, but a vast majority are. However it still lags just as much if I have the namespace hidden. If theres anything I can do to help get you info let me know

**Comment by Nokel81** [comment]
The solution for this is probably related to streaming results to stores. Though the fact that you said that the pods are in the 100s starting and stopping per minute is certainly the bigger problem.

Are you using pods for individual requests? I would like to have some understanding behind why so many pods are being stopped and started. That should help us prioritize this issue.

**Comment by Double0negative** [comment]
We are a game server hosting, so we have lots user requested servers spinning up and down constantly.

**Comment by Nokel81** [comment]
Thanks for the information. I guess we need some feedback.

How "up to date" do you want your information to be if the number of pods is constantly changing?

**Comment by Double0negative** [comment]
I think at that size/rate, the most important thing is being able to find the pod when searching for it. So as long as the search is able to find new pods, it doesn't matter as much if the full list is "live". For smaller, less active namespaces, its far more important for the list to be "live"

**Comment by Nokel81** [comment]
Thanks for the information. I think I will flesh this out into a feature request tomorrow and then close this then.

**Comment by Nokel81** [comment]
@Double0negative Would you be fine if you were required to mark a cluster like this as "large" (or something similar, ignore bikeshedding) and have the system instead of doing streaming we do slow polling and fall back to querying the kube server for an item if we don't find it?

**Comment by Double0negative** [comment]
Yea that would be fine. If possible it would be preferable to do this per namespace, but if not on the whole cluster would be ok

**Comment by Nokel81** [comment]
Per namespace would be interesting, but more difficult when you have both selected, (or for that matter if you select "All Namespaces"). One possibility would be for you to add the cluster twice and limit the namespaces on one (or both). Though that might mean that we should add the opposite of "accessible namespaces" so that you could turn off namespaces that you want to view using the other configuration.

**Comment by hazemarian** [comment]
i am having the same problem here after updating to Lens 5
before it took time to retrieve the cluster info but lens will not hang and stop responding
now lens stop responding and the laptop fans start spin loudly for a few minutes and it will go back to normal again 
the CPU and Memory usage is hitting the roof when this happens 
 `Lens Helper (Renderer) the process that using the resources`

**Comment by Nokel81** [comment]
If you turn off "Allow Error Reporting" in user preferences does Lens's performance improve?

**Comment by hazemarian** [comment]
no it did not i think this problem could be avoided if the default name space is selected by default not all name spaces
also we can retrieve all pods names with out statuses and get the statuses when scrolling down 
i noticed that lest is getting all pods info at one 
we can get list of pod names it will be fast
and create a pagination by 10
each page we retrieve 10 pods info

**Comment by Nokel81** [comment]
The default set of namespaces is current `"default"` or first (if default is not a namespace known to Lens).

The idea about getting events on demand is interesting we will consider it.

**Comment by jakolehm** [comment]
@Double0negative is this still an issue with the latest version (5.3.x)?

**Comment by matti** [comment]
renderer 100%:

<img width="1282" alt="image" src="https://user-images.githubusercontent.com/3893/159356267-d9c24388-130a-448e-ae10-18d251388823.png">

feels like:
https://user-images.githubusercontent.com/3893/159356153-5e70e4ae-4a26-4795-8eda-d1bb6cacf69a.mov
